shell/browser/electron_browser_client.cc

```diff
@@ -1655,6 +1657,15 @@ bool ElectronBrowserClient::ShouldEnableStrictSiteIsolation() {
   return true;
 }
 
+void ElectronBrowserClient::BindUtilityHostReceiver(
+    mojo::GenericPendingReceiver receiver) {
+  if (auto r = receiver.As<node::mojom::NodeServiceHost>()) {
+    static base::NoDestructor<NodeServiceHostImpl> node_service_host;
```

Comment:
this is a little surprising, we just leak this?

Reply:
There is only single instance of this class for the lifetime of the application, hence I went with leaking it.

Reply:
This needs to change to be one host per process. We can't allow the child process to specify its own pid in the `ReceivePostMessage` function.

Reply:
I had a misunderstanding on the contract of the API `content::BrowserClient::BindUtilityHostReceiver`, because it was on the main process global client and only got the receiver endpoint as argument I had assumed it to be 1:many dependency (i-e) One Browser process client gets interface request from multiple utility processes without targeting a specific host. 

But it is actually 1:1 dependency, (i-e) each utility process requests the corresponding process host https://source.chromium.org/chromium/chromium/src/+/main:content/browser/utility_process_host_receiver_bindings.cc;l=20-29 but the host argument does not get forwarded to `content::BrowserClient`, an alternative example is `content::RenderProcessHost` which forwards the host information when interface request comes from render process https://source.chromium.org/chromium/chromium/src/+/main:content/browser/renderer_host/render_process_host_impl.cc;l=5243-5255. I guess there was no exiting utility process in chromium that needed the host details when binding the interface. I will refactor this implementation.